app/src/main/java/net/gsantner/markor/activity/MainActivity.java

```diff
@@ -394,7 +395,8 @@ public void onViewPagerPageSelected(int pos) {
         (_lastBottomMenuItem != null ? _lastBottomMenuItem : menu.getItem(0)).setChecked(false);
         _lastBottomMenuItem = menu.getItem(pos).setChecked(true);
         updateFabVisibility(pos == 0);
-        _toolbar.setTitle(new String[]{_currentTitle, getString(R.string.todo), getString(R.string.quicknote), getString(R.string.linkbox), getString(R.string.more)}[pos]);
+        FilesystemViewerFragment fragment = (FilesystemViewerFragment) _viewPagerAdapter.getFragmentByTag(FilesystemViewerFragment.FRAGMENT_TAG);
+        _toolbar.setTitle(new String[]{fragment.getCurrentFolder().getName(), getString(R.string.todo), getString(R.string.quicknote), getString(R.string.linkbox), getString(R.string.more)}[pos]);
```

Comment:
it may not happen often, but it is possible that FilesystemViewerFragment is null. can you fallback to _currentTitle in that case? (`frag == null ? _currentTitle : getCurrent..`).

Reply:
Thanks for your reply I will make those changes, but while testing I found that the reason for the bug is that `_currentTitle` is never set (so is blank) as lines 246-261 are never run in the BroadcastReceiver. The title is actually set at line 428 when the folder changes.

Instead of changing it to `_currentTitle` if null, should I change it to `getResources().getString(R.string.app_name).toLowerCase()` so that if FilesystemViewerFragment is null, it displays the app name (_markor_) as a backup?

Reply:
sounds good, but you can also fix the cache to save at correct point :D

Reply:
Updates made including a cached folder variable and fallback to app name :)